pkg/channel/message_dispatcher.go

```diff
@@ -191,10 +201,15 @@ func (d *MessageDispatcherImpl) executeRequest(ctx context.Context, url *url.URL
 	}
 
 	if span.IsRecordingEvents() {
-		err = kncloudevents.WriteHTTPRequestWithAdditionalHeaders(ctx, message, req, additionalHeaders, tracing.PopulateSpan(span, url.String()))
-	} else {
-		err = kncloudevents.WriteHTTPRequestWithAdditionalHeaders(ctx, message, req, additionalHeaders)
+		tracingTransformer := tracing.PopulateSpan(span, url.String())
+		if transformers == nil {
+			transformers = []binding.Transformer{tracingTransformer}
+		} else {
+			transformers = append(transformers, tracingTransformer)
+		}
```

Comment:
AFAIK This shouldn't be required, you can safely append to a nil slice

Reply:
yep - thanks!